src/vendored/GPv2Order.sol

```diff
@@ -0,0 +1,255 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+// Vendored from GPv2 contracts v1.1.2, see:
+// <https://raw.githubusercontent.com/cowprotocol/contracts/v1.0.0/src/contracts/libraries/GPv2Order.sol>
+// The following changes were made:
+// - Bumped up Solidity version.
+// - Vendored imports.
+// - Made some constants publicly available.
+
+pragma solidity ^0.8;
+
+import "./IERC20.sol";
+
+/// @title Gnosis Protocol v2 Order Library
+/// @author Gnosis Developers
+library GPv2Order {
+    /// @dev The complete data for a Gnosis Protocol order. This struct contains
+    /// all order parameters that are signed for submitting to GP.
+    struct Data {
+        IERC20 sellToken;
+        IERC20 buyToken;
+        address receiver;
+        uint256 sellAmount;
+        uint256 buyAmount;
+        uint32 validTo;
+        bytes32 appData;
+        uint256 feeAmount;
+        bytes32 kind;
+        bool partiallyFillable;
+        bytes32 sellTokenBalance;
+        bytes32 buyTokenBalance;
+    }
+
+    /// @dev The order EIP-712 type hash for the [`GPv2Order.Data`] struct.
+    ///
+    /// This value is pre-computed from the following expression:
+    /// ```
+    /// keccak256(
+    ///     "Order(" +
+    ///         "address sellToken," +
+    ///         "address buyToken," +
+    ///         "address receiver," +
+    ///         "uint256 sellAmount," +
+    ///         "uint256 buyAmount," +
+    ///         "uint32 validTo," +
+    ///         "bytes32 appData," +
+    ///         "uint256 feeAmount," +
+    ///         "string kind," +
+    ///         "bool partiallyFillable" +
+    ///         "string sellTokenBalance" +
+    ///         "string buyTokenBalance" +
+    ///     ")"
```

Comment:
Nit. There appears to be an awkward inconsistency here where all but the last three lines have commas after them.

Is this literally the hash of a bunch of string concatenations?

Reply:
I think its just a typo from the original cowprotocol/contracts repo (its vendored code, so "copied as is").

> Is this literally the hash of a bunch of string concatenations?

Yes, luckily the hash was computed correctly and the comment is off.

Reply:
Good eyes! I fixed it [here](https://github.com/cowprotocol/contracts/pull/25), as Nick mentions this is just a typo in the comment and has no effect on the computed hash.